isis/src/control/objs/ControlNetVersioner/ControlNetVersioner.h

```diff
@@ -144,18 +162,199 @@ namespace Isis {
+   *   the history of the control network file format prior to ISIS 3.2.2
+   *   is not well documented.
+   *
+   * The following are descriptions of the different control network file
+   *   format versions that are currently supported. Each description also
+   *   describes how that version differs from the previous.
+   *
+   * <b>Version 1</b>
+   *
+   *   This version maintains backwards compatibility with all files created
+   *   prior to versioning. If a control network file does not have a version
+   *   flag, then it is assumed to be a version 1 file. Because this version
+   *   supports all files created prior to ISIS 3.2.2, there is no standardized
+   *   Pvl format associated with it.
+   *
+   *   Originally, there was no version 1 binary format. When version 2 was
+   *   added, version 1 was changed to use the version 2 binary format.
+   *
+   * <b>Version 2</b>
+   *
+   *   This version was the first to have a standardized format. The following
+   *   were standardized with this format:
+   *   <ul>
+   *     <li>The Held flag was replaced by point types. Points could be either
+   *     Tie points or Ground points. Points that were previously flagged as
+   *     Held were changed to Ground points.</li>
+   *     <li>A posteriori was replaced with adjusted in several keyword
+   *     names.</li>
+   *     <li>The a priori and adjusted ground points were changed from
+   *     (latitude, longitude, radius) format to body fixed (X, Y, Z) format.
+   *     </li>
+   *     <li>Ground point sigmas were replaced with covariance matrices.</li>
+   *     <li>Latitude, longitude, and radius constrained flags were added.</li>
+   *     <li>Estimated measures were renamed to Candidate measures.</li>
+   *     <li>Unmeasured measures were renamed to Candidate measures, had their
+   *     line and sample set to 0, and were flagged as ignored.</li>
+   *     <li>Automatic, ValidatedManual and AutomaticPixel measures were
+   *     renamed to RegisteredPixel measures.</li>
+   *     <li>ValidatedAutomatic and AutomaticSubPixel measures were renamed to
+   *     RegisteredSubPixel measures.</li>
+   *     <li>ErrorSample and ErrorLine were renamed to SampleResidual and
+   *     LineResidual respectively.</li>
+   *     <li>Diameter, ZScore, and ErrorMagnitude were no longer saved in
+   *     measures.</li>
+   *   </ul>
+   *
+   *   Version 2 was the first version to support a binary protobuf format.
+   *   Version 1 was retroactively changed to use the version 2 binary format.
+   *   Version 2 binary control network files consist of three parts:
+   *   <ol>
+   *     <li>
+   *     <em>Pvl File Header:</em> The file starts with a Pvl formatted header
+   *     that contains offsets to the binary components of the file. This
+   *     header may also contain general information about the control network.
+   *     This information is only for user reference is not used when reading
+   *     the file.
```

Comment:
Don't we use the offsets to the binary components (from the Pvl header) when reading the file?

Reply:
And I think we also use the version number in the header to determine which readProtobufV#### to use.